orderer/common/follower/follower_chain.go

```diff
@@ -242,20 +210,28 @@ func (c *Chain) StatusReport() (types.ClusterRelation, types.Status) {
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
 
-	clusterRelation := types.ClusterRelationFollower
+	return c.cRel, c.status
+}
+
+func (c *Chain) updateStatusReport() {
```

Comment:
Unfortunately, I don't think this is the solution to the control flow problem.  We're still invoking this status update in three different spots I count in the flow.

What I'm really looking for is a top level control loop, that invokes an action, waits for that action to return, and then sets a status based on the result.

I would expect something that looks like this:

```
targetHeight := c.joinBlock.SeqNo
for {
    if c.ledger.Height() < targetHeight {
        if err := replicateThrough(targetHeight); err != nil {
            // exponential backoff
            continue
        }
        // reset backoff
    }

    // at this point, we are at our replication target, set status
    // check membership, and exit replication if needed

    // at this point, we are still replicating, so wait:
    for {
        networkHeight, err := c.getNetworkHeight()
        if err != nil || networkHeight <= c.ledger.Height() {
            // If we really wanted to be ambitious, we could exponentially backoff in the error case
            time.Sleep(sometime)
            continue
        }

        targetHeight = networkHeight
    }
}
```

Reply:
Addressed in latest commit.